### ISSUE TITLE
Xcode Instruments Memory Allocations does not work with .NET MAUI projects on iOS

### ISSUE THREAD

**Issue by danardelean** [dotnet/maui#20185]
### Description

Debbuging native memory allocations on iOS with Instruments doesn't work for MAUI projects

### Steps to Reproduce

1. Create an .Net MAUI application and deploy on an iOS emulator
2. Try do debug Memory allocations with Instruments The application will run the tool will register a few seconds and then stop
3. If you use a "pure" net8.0-ios project Instruments will not stop tracing the memory allocations

### Version with bug

8.0.6

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

**Comment by danardelean** [comment]
![image](https://github.com/dotnet/maui/assets/1352599/491aa15c-1195-4c7c-b95b-91d12031ead3)

**Comment by danardelean** [comment]
This works for an Net8.0-ios application:
![image](https://github.com/dotnet/maui/assets/1352599/f41beb18-6b23-4eed-a8a9-44c14b5c35d2)

**Comment by verthal** [comment]
hi,
same problem here. We want to use Instruments to trace memory allocations, that are not recorded by dotnet-gcdump or dotnet-trace. It was possible to use Instruments with Xamarin.Forms Apps, but with .Net 8-Maui it doesn't work anymore.

**Comment by danardelean** [comment]
It would be nice to have some kind of feedback from the team not only adding labels

**Comment by AStrickland-abbott** [comment]
I see a similar behavior. If I do new Net-iOS project, the profiler records the whole session. If I try to run on our actual Net-iOS app, which uses Maui Essentials and shared project structure, it only attaches for 2-4 seconds before stopping.
Running 8.0.203 of dotnet, Maui Essentials 8.0.6 and CommunityToolkit.Mvvm 8.2.2

**Comment by Zhanglirong-Winnie** [comment]
Verified this issue with Visual Studio 17.6.12（Build410）and Xcode Instruments (15.1). Can repro this issue.
![image](https://github.com/dotnet/maui/assets/77434865/16ee7665-c645-41a7-a86d-73857c2ea140)

**Comment by danardelean** [comment]
Would be really nice to have more feedback on this issue.

**Comment by RuddyOne** [comment]
Please can we get feedback on this? 

I have tested a new MAUI app running iOS, this doesn't work. Connects for less then 1 second. 

However a .NET iOS app does work just fine. Any workaround for this issue? I have memory leaks I need to find.

**Comment by jmacmullin** [comment]
Is there any plan to address this issue?

Given how many people struggle with memory leaks in MAUI apps then it would be extremely useful if profiling the memory use was simpler.

**Comment by PeWu09** [comment]
Missing that option a lot. Profiling Android is way easier. When it comes to iOS I fell blind.

**Comment by jmacmullin** [comment]
How has this issue been open for a year?

**Comment by UnreachableCode** [comment]
> Missing that option a lot. Profiling Android is way easier. When it comes to iOS I fell blind.

I agree with you. I was able to profile a MAUI app with Android Studio quite easily.

**Comment by rolfbjarne** [comment]
This might be a duplicate of https://github.com/dotnet/runtime/issues/90394.

**Comment by jfversluis** [comment]
Following through that issue it seems like its fixed for .NET 11 and they are backporting it to .NET 10: https://github.com/dotnet/runtime/pull/120293 seeing where we are with .NET 10 I think that will be GA at the earliest where that will land, maybe even a bit later.

To not leave any issues lingering here, I'll close this one for now. If this is important to you, monitor the above issue and PR and when that is released please try again. If that still doesn't resolve it for you, please let us know. Thanks!

**Comment by jmacmullin** [comment]
I’m no longer using MAUI so I’d suggest if this is important to you, monitor the above issue and PR and when that is released please try again.
